lib/hocon/impl/abstract_config_value.rb

```diff
@@ -60,6 +157,78 @@ def require_not_ignoring_fallbacks
     end
   end
 
+  def construct_delayed_merge(origin, stack)
+    # TODO: this might not work because ConfigDelayedMerge inherits
+    # from this class, so we can't `require` it from this file
+    Hocon::Impl::ConfigDelayedMerge.new(origin, stack)
+  end
+
+  def merged_stack_with_the_unmergeable(stack, fallback)
+    require_not_ignoring_fallbacks
+
+    # if we turn out to be an object, and the fallback also does,
+    # then a merge may be required; delay until we resolve.
+    new_stack = stack.clone
+    new_stack.concat(fallback.unmerged_values)
+    # TODO: this might not work because AbstractConfigObject inherits
+    # from this class, so we can't `require` it from this file
+    construct_delayed_merge(Hocon::Impl::AbstractConfigObject.merge_origins(new_stack), new_stack)
+  end
+
+  def delay_merge(stack, fallback)
+    # if we turn out to be an object, and the fallback also does,
+    # then a merge may be required.
+    # if we contain a substitution, resolving it may need to look
+    # back to the fallback
+    new_stack = stack.clone
+    new_stack << fallback
+    # TODO: this might not work because AbstractConfigObject inherits
+    # from this class, so we can't `require` it from this file
+    construct_delayed_merge(Hocon::Impl::AbstractConfigObject.merge_origins(new_stack), new_stack)
+  end
+
+  def merged_stack_with_object(stack, fallback)
```

Comment:
This method, and some others, appear to have different names in the latest version of the upstream code.  For example, this one is `mergedWithObject`.